### ISSUE TITLE
6.1,7.1 channel AAC compliant with ISO/IEC 14496-3:2009/Amendment 4:2013?

### ISSUE THREAD

**Issue by kit90** [nu774/qaac#40]
ISO/IEC 14496-3:2009/Amendment 4:2013 (New levels for AAC profiles) specifies the format of 7.1-channel AAC with rear surrounds. According to it, such a file should have audioProfileLevelIndication = 0x50 (i.e., confirming to AAC Profile Level 6) and ChannelConfiguration = 12. Do the files generated by qaac support this standard?

Also, on a related note, I tried to convert a 7.1-channel WAV file (WAVE_FORMAT_EXTENSIBLE with channel mask 0x63f) to AAC with qaac, fdkaac, Nero AAC and Winamp fhgaac. Quicktime Player displayed the layout of the file encoded with fdkaac as C,L,R,Ls,Rs,Rls,Rrs,LFE in the Movie Inspector window. But for the files encoded with qaac, Nero AAC and fhgaac, it displayed C,Lc,Rc,L,R,Ls,Rs,LFE.

**Comment by nu774** [comment]
>ISO/IEC 14496-3:2009/Amendment 4:2013 (New levels for AAC profiles) specifies the format of 7.1-channel AAC with rear surrounds. According to it, such a file should have audioProfileLevelIndication = 0x50 (i.e., confirming to AAC Profile Level 6) and ChannelConfiguration = 12. Do the files generated by qaac support this standard?

No.  It is a new standard, hence less compatible.
libavcodec didn't support it until this patch: https://lists.libav.org/pipermail/libav-devel/2015-June/069944.html .
 
> Also, on a related note, I tried to convert a 7.1-channel WAV file (WAVE_FORMAT_EXTENSIBLE with channel mask 0x63f) to AAC with qaac, fdkaac, Nero AAC and Winamp fhgaac. Quicktime Player displayed the layout of the file encoded with fdkaac as C,L,R,Ls,Rs,Rls,Rrs,LFE in the Movie Inspector window. But for the files encoded with qaac, Nero AAC and fhgaac, it displayed C,Lc,Rc,L,R,Ls,Rs,LFE.

Both of qaac and fdkaac correctly uses PCE (program config element) to describe 7.1ch rear layout. However, there's a difference. qaac describes it as 3 front + 2 side + 2 back + 1 LFE.
On the other hand, FDK-AAC describes as 3 front + 4 back + 1 LFE. 

As far as I can see, QuickTime recognizes only the latter. On the other hand, libavcodec couldn't recognize the latter until this patch:  https://lists.libav.org/pipermail/libav-devel/2015-June/069932.html 
In other words, there's no way to make everybody happy.

IIRC, nero (and also fhgaac probably) doesn't correctly support 7.1ch rear. It just encode it as 7.1ch front (which is the default AAC layout for 7.1ch).

**Comment by nithin-mk** [comment]
I'm the creator of this issue; my previous GitHub account got deleted. Just like to enquire if this issue could be resolved, as I hope all programs / libraries are compatible with the new standard by now. Thank you! :-)
